src/components/Select/Option.vue

```diff
@@ -1,87 +1,115 @@
 <template>
-  <div
-    @click="choose"
-    @mouseenter="handleFocus"
-    @mouseleave="handleBlur"
-    @focus="handleFocus"
-    @blur="handleBlur"
-    @keydown.enter.space="choose"
-    class="option"
-    :class="{
-      selected: selected,
-      hovered: hovered,
-      disabled: disabled
-    }"
-    tabindex="0"
-  >
-    <span>
-      <i v-if="icon" :class="iconClass" class="pi pi-1x mr-1"></i>
-      <slot>{{ value }}</slot>
-    </span>
-    <i v-if="selected" class="pi pi-check-line pi-lg"></i>
+  <div class="option" :class="classes.option" @click="click">
+    <OptionLabel class="option__label" v-bind="{ icon, label }">
+      <template v-slot:icon="scope">
+        <slot name="icon" v-bind="scope" />
+      </template>
+      <template v-slot:label="scope">
+        <slot name="label" v-bind="scope" />
+      </template>
+    </OptionLabel>
+    <template v-if="isSelected">
+      <i class="pi pi-check-line pi-lg" />
+    </template>
   </div>
 </template>
 
 <script lang="ts">
-import { defineComponent } from 'vue'
+import { Vue, prop } from 'vue-class-component'
+import { Component } from '@/utilities/vue-class-component'
+import OptionLabel from '@/components/Select/OptionLabel.vue'
 
-export default defineComponent({
-  name: 'Option',
-  emits: {
-    click(e: Event, ...args: any[]) {
-      return { e, ...args }
-    }
-  },
-  props: {
-    value: {
-      type: String,
-      required: true,
-      default: 'Option'
-    },
-    data: {
-      type: [String, Object]
-    },
-    icon: {
-      type: String,
-      default: ''
-    },
-    selected: {
-      type: Boolean,
-      default: false
-    },
-    disabled: {
-      type: Boolean,
-      default: false
-    }
-  },
-  data() {
+class Props {
+  label = prop<string>({ required: true })
+  value = prop({ required: true })
+  icon = prop<string>({ default: '' })
+  disabled = prop<boolean | string>({ default: false })
+  selected = prop<boolean | string>({ default: false })
+}
+
+@Component({
+  emits: ['select'],
+  components: {
+    OptionLabel
+  }
+})
+export default class Option extends Vue.with(Props) {
+
+  get isDisabled() {
+    return this.disabled || this.disabled === ''
+  }
+
+  get isSelected() {
+    return this.selected || this.selected === ''
+  }
+
+  get scope() {
     return {
-      hovered: false
+      label: this.label,
+      value: this.value,
+      icon: this.icon,
+      disabled: this.isDisabled,
+      selected: this.isSelected
     }
-  },
-  computed: {
-    iconClass(): string | null {
-      return this.icon.length > 0 ? `pi-${this.icon}` : null
+  }
+
+  get classes() {
+    return {
+      option: {
+        'option--disabled': this.isDisabled,
+        'option--selected': this.isSelected
+      },
+      icon: `pi pi-1x mr-1 pi-${this.icon}`
```

Comment:
possible future enhancement to create our own icon component. It would mean less "pi" classes all over the place and we could confirm that the `icon` prop isn't just a string, but a valid icon in our system

Reply:
yeah that's on my mental list as well. Love the typesafe idea.